### DISCUSSION TITLE
reset should accept an array of input names (the way trigger works)

### DISCUSSION THREAD

**Discussion by conor909** [react-hook-form/react-hook-form#6347] [General]
I could be missing something from the docs here, but I can't just reset a section or individual inputs of my form without resetting the whole form.

[The docs for the `reset` function](https://react-hook-form.com/api/useform/reset) says: 
- You will need to pass `defaultValues` to `useForm` in order to reset the `Controller` components' value.

But how do I tell which `Controller` values to reset, without resetting all of them?  I tried with an array of input names, like how `trigger` works when validating, but it didn't work.

So I tried removing all the `defaultValue` from my `Controller`s.  And added the `defaultValues` to my `useForm` method.  I then tried ways of structuring object values to pass into `reset({ values: {...}})`, but my form has deeply nested objects in some sections and is dynamic and difficult to programmatically create.

I think something like this would be easiest:
```js
reset([
   'person.age',
   'person.hair.color',
   'person.hair.length',
   'person.clothes.coat.color'
])
```

The alternative I'm using at the moment, is passing in `defaultValues` to `useForm` and creating my own `onReset` function from the `defaultValues` object:
```js
  import get from 'lodash.get'

  const onReset = (inputNames:string[]) => {
      inputNames.forEach((name:string) =>
          setValue(name, get(defaultValues, name)))
  }
```

**Comment by Andreychikov-Vasiliy** [comment]
Agree. *setValue* can't always be used because accept 
```typescript 
value: UnpackNestedValue<FieldPathValue<TFieldValues, TFieldName>>
```
so it can't be used because we need provide explicit value, rather that when we need to reset field, we actualy want be able to use something like **setDefaultValue** or **resetTo**
```typescript 
value: UnpackNestedValue<FieldPathDefaultValue<TFieldValues, TFieldName>>
```

**Comment by conor909** [comment]
@bluebill1049 you closed the original issue and marked as completed?  Has this feature been added?
